torch/_dynamo/variables/user_defined.py

```diff
@@ -330,15 +332,23 @@ def call_function(
             and hasattr(
                 self.value, "__exit__"
             )  # TODO(voz): These can invoke user code!
-            and check_constant_args(args, kwargs)
-            and self.value.__init__ == object.__init__
-            and len(kwargs) == 0  # TODO(ybliang): support kwargs
+            # and check_constant_args(args, kwargs)
+            # and self.value.__init__ == object.__init__
+            # and len(kwargs) == 0
+            and self.value.__new__ == object.__new__
+            and SideEffects.cls_supports_mutation_side_effects(self.value)
+            and self.source is not None
         ):
-            unwrapped_args = [x.as_python_constant() for x in args]
-            return GenericContextWrappingVariable(
-                unwrapped_args,
-                cm_obj=self.value(*unwrapped_args),
+            # import here to avoid an unfortunate circular dependency.
+            from .ctx_manager import GenericContextWrappingVariable
+
+            cm_obj = tx.output.side_effects.track_object_new(
+                self.source, self.value, GenericContextWrappingVariable, {}
             )
+            # unwrap constant args
+            cm_obj.target_values = map(torch._dynamo.utils.guard_if_dyn, args)
```

Comment:
@zou3519, this line is new. `target_values` stores arguments passed to the context manager, and it is used in [symbolic_convert.py](https://github.com/pytorch/pytorch/blob/aa7ea6b45cf23db37365edde18498fb98a367f77/torch/_dynamo/symbolic_convert.py#L2540-L2542). Also, if we don't unwrap constant values, pytorch crashes at a later stage.